src/plugins/data_source_management/public/components/data_source_table/use_data_source_updater.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { useCallback } from 'react';
+import { DEFAULT_DATA_SOURCE_UI_SETTINGS_ID } from '../constants';
+import { UiSettingScope } from '../../../../../core/public';
+import { DataSourceTableItem } from '../../types';
+import { WorkspaceObject, IUiSettingsClient, NotificationsStart } from '../../../../../core/public';
+
+interface UseDataSourceUpdaterDeps {
+  fetchDataSources: () => Promise<DataSourceTableItem[]>;
+  defaultDataSourceId: string | null | undefined;
+  uiSettings: IUiSettingsClient;
+  loadDefaultDataSourceId: () => Promise<void>;
```

Comment:
Nit: From the hook perspective, I prefer renaming this props to something like `onDefaultDataSourceIdSet`. `loadDefaultDataSourceId` is a little bit strange to this hook.